### PR TITLE
Shield Generators Make An Alarm When Shutting Down

### PR DESCRIPTION
**Pull request by Erikafox**
## About The Pull Request
As Title.
Requested by Zevo

## Why It's Good For The Game
No more immediate FWOOSH

## Changelog

:cl:
add: Holofans now have a small window before shutting down if they are deactivated by lack of power
/:cl:

### CODE REVIEW

File: code/game/machinery/shieldgen.dm [hunk 1]
```diff
@@ -252,16 +253,25 @@
...
+				crashing = TRUE
+				addtimer(CALLBACK(src, PROC_REF(kill_shield)), 6 SECONDS)
+
 	else
 		for(var/direction in GLOB.cardinals)
 			cleanup_field(direction)
 	update_appearance()
 
+/obj/machinery/power/shieldwallgen/proc/kill_shield()
+	visible_message(span_danger("The [src.name] shuts down due to lack of power!"), "If this message is ever seen, something is wrong.",span_hear("You hear heavy droning fade out.</"))
```
Comment (FalloutFalcon) [inline code/game/machinery/shieldgen.dm]:
```suggestion
	visible_message(span_danger("The [src.name] shuts down due to lack of power!"), "If this message is ever seen, something is wrong.",span_hear("You hear heavy droning fade out."))
```

**Review by FalloutFalcon — APPROVED** [review]
Looks good, honestly something like 10 seconds would be fine as well. The thing has a short backup power supply because of how often people fucking space their ships.

**Comment by Apogee-dev** [comment]
I’m gonna throw in what I’ve usually mentioned before when talking about fixing holofans (thank you for being the one to do it btw): Please make sure they’re still forcibly disrupted if the ship docks/undocks with a holofield active. Smoothing over power disruptions and recalculations is good, but taking off with your doors open is OSHA safety video material

**Comment by Zevotech** [comment]
That behavior should be hardcoded to happen instead of being a consequence of the powernet updating, honestly.

**Pull request merged**